### PR TITLE
Firebase analytics - Dashboard

### PR DESCRIPTION
**Pull request by Sashen943**
- Analytic functions added to presenter
- Implemented and injected through view controller
- Unit tests written for presenter 
- General code clean up of dashboard feature:
> Renaming variables
> Removed white spaces
> Added and improved MARK comments

### CODE REVIEW

File: Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift [hunk 1]
```diff
@@ -33,33 +34,40 @@ class HomePresentableTests: XCTestCase {
         homePresenter.firebaseStorage = mockFirebaseStorage
         homePresenter.signOutInteractor = mockSignOutInteractor
         homePresenter.userDefaults = mockUserDefaults
-        systemUnderTest = homePresenter
+        homePresenter.analyticManager = mockAnalyticsManager
+        presenterUnderTest = homePresenter
+    }
+
+    override func tearDown() {
```
Comment (pjwelcome) [inline Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift]: Please remove the tear down if you are not using it for anything

File: Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift [hunk 2]
```diff
@@ -33,33 +34,40 @@ class HomePresentableTests: XCTestCase {
         homePresenter.firebaseStorage = mockFirebaseStorage
         homePresenter.signOutInteractor = mockSignOutInteractor
         homePresenter.userDefaults = mockUserDefaults
-        systemUnderTest = homePresenter
+        homePresenter.analyticManager = mockAnalyticsManager
+        presenterUnderTest = homePresenter
```
Comment (pjwelcome) [inline Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift]: why did you rename this to presenterUnderTest? systemUnder test is a more universal term for unit testing so that we know what the class we are testing. Please revert the renaming of systemUnderTest

**Review by pjwelcome — CHANGES_REQUESTED** [review]

File: Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift (hunk 2 above)
Comment (Sashen943) [reply Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift]: done

File: Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift (hunk 1 above)
Comment (Sashen943) [reply Showcase-iOSTests/Dashboard/Presenter/HomePresentableTests.swift]: done :)

**Review by pjwelcome — APPROVED** [review]
